docs/application/native/guides/app-management/tidl.md

```diff
@@ -0,0 +1,328 @@
+
+# TIDL
+
+TIDL is a programming language to define interfaces for communicating among apps in Tizen.
+It provides methods to create a Remote Procedure Call (RPC) or Remote Method Invocation (RMI) in Tizen.
+
+## TIDLC
+
+**TIDLC** is a compiler for creating stub or proxy code block from **TIDL** file.
+
+### Usage
+```
+Usage:
+  tidlc [OPTION...]
+
+  -h, --help                  Show help options.
+  -l, --language=LANGUAGE     Select generating language (C, C++, C#).
+  -i, --input=INPUT           A tidl interface file.
+  -o, --output=OUTPUT         The generated interface file.
+  -p, --proxy                 Generate proxy code.
+  -s, --stub                  Generate stub code.
+  -v, --version               Show version information.
+
+```
+
+## TIDL Syntax
+
+### 'interface'
+It makes an interface to communicate between proxy and stub.
+
+**Syntax**
+```csharp
+[<attributes>]
+interface <interface_id> {
+<return_type> <method_name>(
+[<empty> | ‘in’ | ‘out’ | ‘ref’]
+<type> <parameter_id>, …) [<empty> | ‘async’ | ‘delegate’\];
+…
+}
+```
+**Example**
+```csharp
+[privilege=“http://tizen.org/privilege/appmanager.launch”]
+interface ITest {
+	// by default, ‘in’ is used as direction keyword
+	int SetVal(in  int val1, char val2);
+	char GetVal(out  int val1, out  char val2);
+}
+```
+
+### Attributes
+It indicates required privileges and trusted signature.
+By default, ‘trusted” is ‘false’.
+
+**Syntax**
+```
+[ <key> = <value>, …]
+interface <interface_id> { … }
+<key> :=  privilege | trusted
+<value> := <string> | “true” | “false”
+```
+
+**Example**
+```csharp
+// Require launch privilege
+[privilege=“http://tizen.org/privilege/appmanager.launch”]
+interface ITest {}
+
+// Require privilege A , privilege B and trusted signature
+[privilege=“http://samsung.com/appdefined/A”,
+ privilege=“http://samsung.com/appdefined/B” , trusted  = “true”]
+interface ITest2 {}
+```
+
+### 'async'
+It denotes one-way-call.
+
+>**Note**
```

Comment:
Please change note like below:

```
> **Note**
>
> Returned type should be `void`. 
>
> Keyword **[out]** is not allowed
```

Please confirm [out] is correct.  I'm not sure which one is correct both [out] and out.

Reply:
I think `out` is correct